testkit/appkit/app.go

```diff
@@ -93,41 +92,75 @@ func (ak *AppKit) GetNodeId(home string) (string, error) {
 	return ak.execCmd([]string{"tendermint", "show-node-id", "--home", home})
 }
 
-func (ak *AppKit) StartNode(home string) error {
+func (ak *AppKit) StartNode(home, loglvl string) error {
 	ak.Cmd.ResetFlags()
 
 	ak.Cmd.SetErr(os.Stdout)
-	ak.Cmd.SetArgs([]string{"start", "--home", home, "--log_level", "info"})
+	ak.Cmd.SetArgs([]string{"start", "--home", home, "--log_level", loglvl, "--log_format", "plain"})
 
 	return svrcmd.Execute(ak.Cmd, appcmd.EnvPrefix, app.DefaultNodeHome)
 }
 
-func GetBlockHashByHeight(ip net.IP, height int) (string, error) {
-	uri := fmt.Sprintf("http://%s:26657/block?height=%d", ip.To4().String(), height)
+func (ak *AppKit) PayForData(accAdr string, msg int, krbackend, chainId, home string) error {
+	ak.Cmd.ResetFlags()
+	ak.Cmd.SetArgs([]string{
+		"tx", "payment", "payForData", fmt.Sprint(msg),
+		"--from", accAdr, "-b", "block", "-y", "--gas", "1000000000",
+		"--fees", "100000000000utia", //15
+		"--keyring-backend", krbackend, "--chain-id", chainId, "--home", home, "--keyring-dir", home,
+	})
+
+	return svrcmd.Execute(ak.Cmd, appcmd.EnvPrefix, app.DefaultNodeHome)
+}
+
+func getResultBlockResponse(uri string) (coretypes.ResultBlock, error) {
```

Comment:
you can make `(*coretypes.ResultBlock, error)`, so you will return nil in case of error instead of creating an empty struct.

Reply:
done! Thanks for the hint!